server/endpoints.ts

```diff
@@ -0,0 +1,116 @@
+import express from "express";
+import { body, validationResult, ValidationChain } from "express-validator";
+import { Classes } from "./dbDefs";
+
+// A type which captures an endpoint, and the guard for that endpoint
+// INVARIANT: If an object passes the guard, it can be coerced into type T
+export interface Endpoint<T> {
+    guard: ValidationChain[];
+    callback: (args: T) => any;
+}
+
+/*
+ * Configure the various endpoints to use
+ */
+export function configure(app: express.Application) {
+  const methods = express.Router();
+  methods.use(express.json());
+  app.use(express.json());
+
+  register(app, "getClassesByQuery", getClassesByQuery);
+}
+
+function register<T>(app: express.Application, name: string, endpoint: Endpoint<T>) {
```

Comment:
Maybe this has to do with the shim or using Typescript with Express, but why not use app.use("/v2")? When I've used Express without TS that was sufficient for directing requests to a set of endpoints.

Reply:
I think that this is more explicit about where all the new endpoints are listening. We can consider changing it though.